src/histogram.js

```diff
@@ -259,9 +266,20 @@ angular.module('grafana.directives').directive('grafanaHistogram', function($roo
           }
         };
 
+        var scopedVars = ctrl.panel.scopedVars;
+        var bucketSize = !panel.bucketSize && panel.bucketSize !== 0 ? null : parseFloat(ctrl.templateSrv.replaceWithText(panel.bucketSize.toString(), scopedVars));
+        var minValue = !panel.minValue && panel.minValue !== 0 ? null : parseFloat(ctrl.templateSrv.replaceWithText(panel.minValue.toString(), scopedVars));
+        var maxValue = !panel.maxValue && panel.maxValue !== 0 ? null : parseFloat(ctrl.templateSrv.replaceWithText(panel.maxValue.toString(), scopedVars));
+
+        switch(panel.bucketMode) {
+          case 'count':
+            bucketSize = (maxValue - minValue) / bucketSize;
```

Comment:
added check for negative values in https://github.com/mtanda/grafana-histogram-panel/pull/29/commits/9513674a575ed2939da52e3df40354d086c5f32b